extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -450,6 +452,15 @@ export class ProjectsController {
 					this.reloadProject(projectFileUri);
 				}
 			});
+
+			// stop watching for changes to the sqlproj after it's closed
+			const closeSqlproj = vscode.workspace.onDidCloseTextDocument((d) => {
```

Comment:
What happens if the sqlproj file is still open, but the project has been closed and removed from the explorer viewlet?

Reply:
fixed to remove watcher when project is closed